client/js/analysis.js

```diff
@@ -1,25 +1,113 @@
+const COMMENT_APPEARANCE_TIME = 1500;
+const COMMENT_TO_STOP_AT = 5;
+
 window.onload = function() {
-  for (let i = 1; i < 6; i++) {
+  // Listen for submission click
+  const formSubmit = document.getElementById('sendbutton');
+  formSubmit.addEventListener('click', fetchResponse, false);
+};
+
+/**
+ * Fetches the data from the backend and populates the screen
+ *
+ */
+async function fetchResponse() {
+  const videoQuery = document.getElementById('searchbar').value;
+  const response = await fetch(`/api/analysis?name=${videoQuery}`);
+  const responseJson = await response.json();
+  populationHandler(responseJson);
+}
+
+/**
+ * Handles populating all elements on the screen
+ *
+ * @param {object} videoAnalysis the object returned from the HTTP request
+ * which contains all of the data.
+ */
+function populationHandler(videoAnalysis) {
+  const charts = document.getElementById('charts');
+  const list = document.getElementById('list');
+
+  removeAllChildNodes(charts);
+  removeAllChildNodes(list);
+  populateDonutCharts(videoAnalysis.perspectiveMap);
+  // The +1 allows for determineOverall() to run correctly.
+  const time = renderComments(videoAnalysis.commentArray) + 1;
+  const overall = determineOverall(videoAnalysis.magnitudeAndScore.magnitude,
+      videoAnalysis.magnitudeAndScore.score);
+  setTimeout(() => {
+    addFeedbackResult(overall);
+  }, time*COMMENT_APPEARANCE_TIME);
+}
+
+/**
+ * Puts the charts on the screen
+ *
+ * @param {Map<string, number>} map the map from the videoAnalysis object
+ */
+function populateDonutCharts(map) {
+  for (const key in map) {
+    if (Object.prototype.hasOwnProperty.call(map, key)) {
+      // Convert to string for type compatibility
+      addDonutChart(key, map[key]);
+    }
+  }
+}
+
+/**
+ * Puts the comments on the screen
+ *
+ * @param {Array} array the comment array from the json response
+ * @returns {number} the stop index so that overall knows when to come in
```

Comment:
rephrase this comment to something like "the total number of comments rendered", this function doesn't depend on its caller so "overall" is out of context